src/DataStructures.jl

```diff
@@ -94,4 +94,6 @@ module DataStructures
     export status
     export deref_key, deref_value, deref, advance, regress
 
+    @deprecate iter(s::Stack) collect(s)
+    @deprecate iter(q::Queue) collect(q)
```

Comment:
It makes sense to me that the deprecation leads you to similar functionality (`collect`). Worst case scenario, the developer has something a little less efficient that is well documented elsewhere.

The alternative (just suggesting `s` or `q`) is that the developer is directed to something that likely wouldn't work in some scenarios.

That is, unless there is a way to provide more descriptive deprecation warnings...